### PR TITLE
NRFX-6372: Automate testing for systemoff sample - wakeup from pin

### PR DESCRIPTION
**Pull request by nordic-pikr**

### CODE REVIEW

File: scripts/ci/tags.yaml [hunk 1]
```diff
@@ -1240,6 +1240,7 @@ ci_tests_benchmarks_current_consumption:
   files:
     - modules/hal/nordic/nrfx/
     - nrf/tests/benchmarks/current_consumption/
+    - zephyr/drivers/gpio
```
Comment (nordic-piks) [inline scripts/ci/tags.yaml:1243]: I will conflict with https://github.com/nrfconnect/sdk-nrf/pull/17559, maybe you can skip that update?

File: tests/benchmarks/current_consumption/system_off/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(system_off)
+
+target_sources(app PRIVATE ../../../../../zephyr/samples/boards/nrf/system_off/src/main.c)
```
Comment (nordic-piks) [inline tests/benchmarks/current_consumption/system_off/CMakeLists.txt:11]: Maybe base on ZEPHYR_BASE env of any other cmake variable with path to zephyr?

**Review by nordic-bami — APPROVED** [review]

File: tests/benchmarks/current_consumption/system_off/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
+    integration_platforms:
+      - nrf54l15dk/nrf54l15/cpuapp
+    extra_args: DTC_OVERLAY_FILE="boards/nrf54l15dk_nrf54l15_cpuapp_gpio_wakeup.overlay"
+    harness: pytest
+    harness_config:
+      fixture: ppk_power_measure
+      pytest_root:
+        - ../test_measure_power_consumption.py::test_measure_and_data_dump_power_consumption_with_wakeups"
```
Comment (nordic-piks) [inline tests/benchmarks/current_consumption/system_off/testcase.yaml:16]: The path should be like: ${CUSTOM_ROOT_TEST_DIR}/test_measure_power_consumption.py:...

File: tests/benchmarks/current_consumption/system_off/prj.conf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+CONFIG_PM_DEVICE=y
+CONFIG_GPIO=y
+CONFIG_CRC=y
```
Comment (nordic-piks) [inline tests/benchmarks/current_consumption/system_off/prj.conf:3]: CRC is from original sample?

Comment (nordic-pikr) [reply tests/benchmarks/current_consumption/system_off/prj.conf:3]: yes; https://github.com/nrfconnect/sdk-zephyr/blob/main/samples/boards/nrf/system_off/prj.conf#L3

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/current_consumption/system_off/Kconfig [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+source "Kconfig.zephyr"
```
Comment (nordicjm) [inline tests/benchmarks/current_consumption/system_off/Kconfig:7]: get rid of this file?

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: tests/benchmarks/current_consumption/system_off/Kconfig (hunk 5 above)
Comment (nordic-pikr) [reply tests/benchmarks/current_consumption/system_off/Kconfig]: removed

File: tests/benchmarks/current_consumption/system_off/Kconfig.sysbuild [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+source "${ZEPHYR_BASE}/share/sysbuild/Kconfig"
```
Comment (nordicjm) [inline tests/benchmarks/current_consumption/system_off/Kconfig.sysbuild:7]: also this file to go, it does nothing

**Review by nordicjm — APPROVED** [review]

**Review by PerMac — APPROVED** [review]

**Pull request merged**